lib/connect/ws.js

```diff
@@ -24,11 +34,15 @@ function buildBuilder (client, opts) {
     wsOpt.protocol = 'mqttv3.1'
   }
 
-  if (opts.wsOptions) {
-    return websocket(url, [wsOpt.protocol], opts.wsOptions)
-  } else {
-    return websocket(url, wsOpt)
+  if (opts.protocol === 'wss') {
```

Comment:
This feels like a common use case to me. I'd do:

```js
var select = require('select-keys')
opts.wsOptions = select(opts, wssProperties)
```

See https://github.com/mmalecki/select-keys/blob/master/lib/select-keys.js